scheduler/util.go

```diff
@@ -455,6 +455,15 @@ func setStatus(logger *log.Logger, planner Planner,
 func inplaceUpdate(ctx Context, eval *structs.Evaluation, job *structs.Job,
 	stack Stack, updates []allocTuple) (destructive, inplace []allocTuple) {
 
+	// doInplace manipulates the updates map to make the current allocation
+	// an inplace update.
+	doInplace := func(cur, last, inplaceCount *int) {
```

Comment:
I disagree. It is moving the allocation to the portion of the slice that is being returned as inplace updates and incrementing the in-place count. All its side effects are to make the current alloc an inplace update.